src/bundle/Resources/public/js/scripts/fieldType/ezdatetime.js

```diff
@@ -91,13 +94,24 @@
         const sourceInput = field.querySelector(SELECTOR_INPUT);
         const flatPickrInput = field.querySelector(SELECTOR_FLATPICKR_INPUT);
         const btnClear = field.querySelector('.ez-data-source__btn--clear-input');
-        const defaultDate = sourceInput.value ? new Date(sourceInput.value * 1000) : null;
+        const secondsEnabled = sourceInput.dataset.seconds === '1';
+        const formatDate = secondsEnabled ? (date) => date.toLocaleString() : (date) => eZ.helpers.timezone.formatDate(date);
```

Comment:
Note: As agreed with @SylvainGuittard, if seconds are enabled we fall back to `date.toLocaleString()` because there it high probability that date format from user settings does not include seconds.